### PR TITLE
refactor: class to edit message_context

### PR DESCRIPTION
**Pull request by xygyo77**
## Description

AssignContextReader is a class that inherits from ArchitectureReader, but it implements almost none of ArchitectureReader's functions, making the code difficult to understand.
Delete the AssignContextReader class and change the implementation to be more in line with the actual situation.

## Related links

[https://tier4.atlassian.net/browse/RT2-1845](https://tier4.atlassian.net/browse/RT2-1845)

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] (Optional) The PR has been properly tested with CARET_report verification.
- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/architecture/architecture.py [hunk 1]
```diff
@@ -762,45 +765,6 @@ def remove_callback_chain(
     def get_message_contexts(self, _) -> Sequence[dict]:
```
Comment (ymski) [inline src/caret_analyze/architecture/architecture.py:765]: Is the argument `_` required?

**Comment by ymski** [comment]
@xygyo77 
Could you create a test for the ContextUpdater class?

File: src/caret_analyze/architecture/architecture.py (hunk 1 above)
Comment (xygyo77) [reply src/caret_analyze/architecture/architecture.py:765]: I agree. The “node” argument of get_message_contexts() is not used, so it is removed.

**Comment by xygyo77** [comment]
> @xygyo77 Could you create a test for the ContextUpdater class?

Since it is an alternative to the AssignContextReader class, I assume it is covered in caret_analyze/src/test/architecture/test_architecture.py.
* test_update_message_context()
* test_remove_variable_passing()

**Comment by ymski** [comment]
> > @xygyo77 Could you create a test for the ContextUpdater class?
> 
> Since it is an alternative to the AssignContextReader class, I assume it is covered in caret_analyze/src/test/architecture/test_architecture.py.
> 
> * test_update_message_context()
> * test_remove_variable_passing()

The above test case was able to test for this change. It seems that we don't need to add any additional tests. Thanks.

**Review by ymski — APPROVED** [review]
LGTM

**Review by miyakoshi-dev — APPROVED** [review]
[compare] ========== Result ==========
[compare] 1/3 file existence matches
[compare] OK. file existence matches
[compare] 2/3 yaml files
[compare] OK. yaml matches
[compare] 3/3 PNG files
[compare] OK. PNG matches
[compare] ====== ALL OK ==============
LGTM

**Pull request merged**